File: generated-code/swift/codecraft/TcpReadWrite/Sources/TransGenTest/Vec2Int.swift
/// 2 dimensional vector.
public struct Vec2Int {
    /// `x` coordinate of the vector
    let x: Int32

    /// `y` coordinate of the vector
    let y: Int32

    /// Read Vec2Int from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Vec2Int {
        var x: Int32
        x = stream.readInt32()
        var y: Int32
        y = stream.readInt32()
        return Vec2Int(x: x, y: y)
    }

    /// Write Vec2Int to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(x)
        stream.writeInt32(y)
    }
}

File: generated-code/php/codecraft/TcpReadWrite/Vec2Int.php
<?php

namespace  {
    require_once 'Stream.php';

    /**
     * 2 dimensional vector.
     */
    class Vec2Int
    {
        /**
         * `x` coordinate of the vector
         */
        public int $x;
        /**
         * `y` coordinate of the vector
         */
        public int $y;
    
        function __construct(int $x, int $y)
        {
            $this->x = $x;
            $this->y = $y;
        }
    
        /**
         * Read Vec2Int from input stream
         */
        public static function readFrom(\InputStream $stream): Vec2Int
        {
            $x = $stream->readInt32();
            $y = $stream->readInt32();
            return new Vec2Int($x, $y);
        }
        
        /**
         * Write Vec2Int to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32($this->x);
            $stream->writeInt32($this->y);
        }
    }
}